### DISCUSSION TITLE
How to apply global phase to a circuit in tequila

### DISCUSSION THREAD

**Discussion by ACE07-Sev** [tequilahub/tequila#365] [Q&A]
Greetings there,

Hope all are well. I have a quick question. May I ask how I can add a global phase gate to the circuit?

**Comment by kottmanj** [comment]
Hi,

in general global phase gate do basically not exist. Tequila actually exploits this: 
E.g. if you initialize a gate like:  
```math
e^{-i\frac{a}{2} \left( 1 + X \right) }
```

```python
U = tq.gates.Trotterized(generator="1.0+X(0)", angle="a")
```
then tequila will internally recompile when the quantum backend is called (not directy as for example `U.add_controls([...])` would be affected) which leaves you with essentially

```math
e^{-i\frac{a}{2} \left( 1 + X \right) } \rightarrow e^{-i\frac{a}{2} \left( X \right) }
```

Can you give a more detailed explanation about the scenario where you want a global phase operation?

**Comment by ACE07-Sev** [comment]
Greetings dear Professor. Kottman,

Hope you are well. Thank you very much for the prompt response!

So, I use the global phase in state preparation and unitary preparation to ensure that the resulting circuit is equal to the statevector and unitary matrix mentioned. Essentially, I provide it to correct any algorithm that picks up a global phase.

**Comment by ohuettenhofer** [comment]
Support for this was recently merged in https://github.com/tequilahub/tequila/pull/417, however there is likely still code that doesn't properly handle global phases (feel free to open an issue if you have code where this happens).
